### PR TITLE
[TOD-2] : Created a REST API for delete a task by ID

### PR DESCRIPTION
**Pull request by jackupsurya**

### CODE REVIEW

**Review by Ajithkumar-25 — APPROVED** [review]

File: http/router.go [hunk 1]
```diff
@@ -12,5 +12,6 @@ func Router(app *gin.Engine) {
 	app.GET(viper.GetString(c.GetAllTasksEndpoint), handler.GetAllTasks)
 	app.GET(viper.GetString(c.GetTaskByIdEndpoint), handler.GetTaskById)
 	app.POST(viper.GetString(c.CreateTaskEndpoint), handler.CreateTask)
-	app.PATCH(viper.GetString(c.UpdateTaskEndpoint), handler.UpdateTask)
+	app.PATCH(viper.GetString(c.UpdateTaskEndpoint), handler.UpdateTaskById)
```
Comment (dev-marees) [inline http/router.go:15]: Why did you use patch here?

Comment (jackupsurya) [reply http/router.go:15]: Bcz Postgrest uses the patch method for updates. That's why, here I'm using it

Comment (dev-marees) [reply http/router.go:15]: What's the difference between the put and patch?

Comment (jackupsurya) [reply http/router.go:15]: To my knowledge, PUT method is used to update a entire record. But PATCH method is used to update a partial record.

**Review by dev-marees — APPROVED** [review]

**Pull request merged**